### DISCUSSION TITLE
x[-1] in init:

### DISCUSSION THREAD

**Discussion by traderblakeq** [kernc/backtesting.py#771] [General]
Hi, - I would like to calculate a indicator, based on previous "[-1]" value:

<img width="550" alt="image" src="https://user-images.githubusercontent.com/36839989/194720416-c0baff3a-a262-4027-933b-7aabc68fd466.png">

(Line 15) - Does not work.

Any inputs ?. ( I know I still need to learn the data stucture)... learning every day :)

Thanks!

Error Stack:

<img width="820" alt="image" src="https://user-images.githubusercontent.com/36839989/194720683-8693bcf5-d68c-4381-bc38-278c7904ccaa.png">

**Comment by kernc** [comment]
You're looking for [`pd.Series.shift()`](https://pandas.pydata.org/docs/reference/api/pandas.Series.shift.html). Need to keep that return value a vector (rather than the last scalar) for `Strategy.I()`.
```py
self.prev_wr14_sig = self.I(lambda: self.wr14.s.shift() > -50)
```

**Reply by traderblakeq** [reply]
Have tried .shift() - But apparantly the Strategy.I() do not have that method ?.

<img width="899" alt="image" src="https://user-images.githubusercontent.com/36839989/194864337-675965d0-4826-49fe-8b5d-f65652e06584.png">

**Reply by traderblakeq** [reply]
This worked:

`        self.wr14_prev = self.I(lambda: self.wr14.s.shift() > -50)
`

**Reply by kernc** [reply]
Apologies. Obviously, you have to cast to `pd.Series` to use its API.
